### PR TITLE
nextcloud: 19.0.1 -> 19.0.3, deprecate/remove older versions

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Closes #97751. 

ChangeLogs:

* https://nextcloud.com/changelog/#19-0-3
* https://nextcloud.com/changelog/#18-0-9

Also removes `nextcloud17` and deprecates `nextcloud18` since both will be EOLed within the lifetime of 20.09[1]. While `nextcloud17` can be removed entirely now, `nextcloud18` must stay (and marked as insecure) to make sure that users from Nextcloud17 can seamlessly upgrade to `nextcloud19` on 20.09.

[1] https://docs.nextcloud.com/server/19/admin_manual/release_schedule.html

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by bachp — APPROVED** [review]

**Comment by Ma27** [comment]
cc @worldofpeace @jonringer I'd consider this and #98332 relevant for 20.09 :)

**Comment by worldofpeace** [comment]
> cc @worldofpeace @jonringer I'd consider this and #98332 relevant for 20.09 :)

I guess this should be a blocker then

**Review by jonringer — APPROVED** [review]
diff LGTM
tests pass

```
https://github.com/NixOS/nixpkgs/pull/98333
1 package marked as broken and skipped:
nextcloud18

1 package blacklisted:
tests.nixos-functions.nixos-test

1 package built:
nextcloud19
```

**Comment by Ma27** [comment]
@worldofpeace anything else to add? I'd consider this good to go and would backport after merge :)

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (worldofpeace) [inline nixos/doc/manual/release-notes/rl-2009.xml:841]: how about `will be EOL (end of life) within the timeline of 20.09`?

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (worldofpeace) [inline nixos/doc/manual/release-notes/rl-2009.xml:844]:
```suggestion
     It's necessary to upgrade to <package>nextcloud19</package>:
```

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (worldofpeace) [inline nixos/doc/manual/release-notes/rl-2009.xml:849]:
```suggestion
        Nextcloud doesn't allow going multiple major revisions forward in a single upgrade. This is possible
```

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (worldofpeace) [inline nixos/doc/manual/release-notes/rl-2009.xml:856]: shouldn't this be `to nextcloud19`?

**Comment by worldofpeace** [comment]
I had @cole-h look at the release note and he pre-approved on IRC with my suggested changes.

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (cole-h) [reply nixos/doc/manual/release-notes/rl-2009.xml:841]: Even just `s/get/be/` would make this sound less awkward.

**Comment by Ma27** [comment]
@worldofpeace @cole-h thanks, fixed :)

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (cole-h) [inline nixos/doc/manual/release-notes/rl-2009.xml:848]:
```suggestion
        From <package>nextcloud17</package>, you have to upgrade to <package>nextcloud18</package> first as
```

File: nixos/doc/manual/release-notes/rl-2009.xml
Comment (cole-h) [inline nixos/doc/manual/release-notes/rl-2009.xml:855]:
```suggestion
        From <package>nextcloud18</package>, it's possible to directly upgrade to <package>nextcloud19</package>
```

**Review by cole-h — APPROVED** [review]
2 more nits, and then diff LGTM.

**Comment by Ma27** [comment]
@GrahamcOfborg eval

**Review by cole-h — APPROVED** [review]
Thanks!

**Comment by Ma27** [comment]
Backported to 20.09 as 5806cedcc1b09582511284f679ed5eb696e44917, 3e771553a38b515ad9ea8072e59a63c2d3237521, c57d809ef5181f984ef5c9613bad9b4a1c30ca30.

Package updates are also on 20.03 from #98332.

**Pull request merged**
